File: the_maps-jigsaw/larstvei.py
def minedges(n):
    i = int(n**0.5)
    while i > 1:
        if n % i == 0:
            break
        i -= 1
    return (i, n/i)

def edgepieces((n,m)):
    if n == 1:
        return m
    return 2*n + 2*m - 4

print edgepieces(minedges(8387035452956160000))

File: the_maps-jigsaw/larstvei.hs
isqrt = floor . sqrt . fromIntegral

divides n = (==0) . (mod n)

minedges n = (i, n `quot` i)
    where m = isqrt n
          i = head $ filter (divides n) [m,m-1..1]

edgepieces (n, m) | n == 1 = m
                  | otherwise = 2*n + 2*m - 4

main = print $ edgepieces $ minedges 8387035452956160000
